proposals/infra/INF-0004-inproc-outofproc-compiler-library-support.md

```diff
@@ -0,0 +1,138 @@
+<!-- {% raw %} -->
+
+# In-process only or In-process and Out-of-process support for the compiler library
+
+* Proposal: [0004](INF-0004-inproc-outofproc-compiler-library-support.md)
+* Author(s): [Cooper Partin](https://github.com/coopp)
+* Sponsor: [Cooper Partin](https://github.com/coopp)
+* Status: **Under Consideration**
+* Impacted Project(s): (Clang)
+
+* Issues:
+
+## Introduction
+
+An effort is underway to bring compilation support into clang for HLSL based
+shaders.  A C style library will be built to enable applications to compile
+a shader from their own processes in addition to being able to launching the
+clang process.
+
+There are two proposed ways one could consume the shader compiler library and
+compile a shader today.
+
+* **In Process** - An application links against the shader compiler library to
+bring in support for shader compilation as a static library
+(full implementation) or a dynamic library. Compilation is performed within the
+calling application's process.
+
+The DirectX Shader library does not currently support being linked as a
+static library.  It is a COM interface library that brings in compilation
+support dynamically. Both static and dynamic in-proc solutions will be 
+supported in the HLSL clang based library.
```

Comment:
> Both static and dynamic in-proc solutions will be supported in the HLSL clang based library.

Proposing that the new API will support static and dynamic is a pretty big decision being made here and so shouldn't be tucked away in the introduction.

Reply:
Made the proposal actually 'propose' a solution rather than be wishy washy about all possible options.